### PR TITLE
tiltfile: make a link object a struct

### PR DESCRIPTION
**Pull request by nicks**
Hello @maiamcc, @landism,

Please review the following commits I made in branch nicks/struct:

eee98cf3b6efe0603035287c5fdbfb9716958cf6 (2020-10-08 16:37:01 -0400)
tiltfile: make a link object a struct

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: internal/tiltfile/links/links.go [hunk 1]
```diff
@@ -117,6 +101,10 @@ func (e Extension) link(thread *starlark.Thread, fn *starlark.Builtin, args star
 	}
 
 	return Link{
+		Struct: starlarkstruct.FromStringDict(starlarkstruct.Default, starlark.StringDict{
```
Comment (landism) [inline internal/tiltfile/links/links.go:104]:
```suggestion
		Struct: starlarkstruct.FromStringDict(starlark.String("link"), starlark.StringDict{
```

I don't know why the code calls the first arg "constructor", but it looks like it acts as the type. It shows up in `.String()` and is compared when checking if two structs are equal or can be added (apparently structs support addition!?)

File: internal/tiltfile/links/links.go [hunk 2]
```diff
@@ -117,6 +101,10 @@ func (e Extension) link(thread *starlark.Thread, fn *starlark.Builtin, args star
...
+			"url":  starlark.String(withScheme),
+			"name": starlark.String(name),
+		}),
 		Link: model.Link{
```
Comment (landism) [inline internal/tiltfile/links/links.go:108]: It feels a bit awkward to have this data in here twice - maybe it'd be less awkward to remove this field and have `LinkList.Unpack` convert from `links.Link` to `model.Link`?

**Review by landism — APPROVED** [review]

File: internal/tiltfile/links/links.go (hunk 1 above)
Comment (nicks) [reply internal/tiltfile/links/links.go]: done

File: internal/tiltfile/links/links.go (hunk 2 above)
Comment (nicks) [reply internal/tiltfile/links/links.go:108]: i played around with this and the conversion logic from a generic Struct to a model.Link would be complicated/error-prone (i.e., the struct interface isn't really made for this kind of conversion), so i'm not sure that'd be less awkward

**Review by maiamcc — COMMENTED** [review]
nice, ty for this!

**Pull request merged**